Broken on 5.8.8

Yes, I know 5.8.8 is old.  But there are a lot folks still using it for various reasons.

In any case, `cpan install Module::Starter` fails as follows:

```
  XSAWYERX/Module-Starter-1.71.tar.gz
  /usr/bin/make -- OK
Running make test
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-load.t ......... 1/4 # Testing Module::Starter 1.71, Perl 5.008008, /usr/bin/perl
t/00-load.t ......... ok
t/BuilderSet.t ...... ok
t/module-starter.t .. skipped: these tests must be completely rewritten
t/pod-coverage.t .... skipped: these tests are for release candidate testing
t/pod.t ............. skipped: these tests are for release candidate testing
t/test-dist.t ....... Added to MANIFEST: Build.PL
Added to MANIFEST: Changes
Added to MANIFEST: ignore.txt
Added to MANIFEST: lib/MyModule/Test.pm
Added to MANIFEST: lib/MyModule/Test/App.pm
Added to MANIFEST: MANIFEST
Added to MANIFEST: README
Added to MANIFEST: t/00-load.t
Added to MANIFEST: t/manifest.t
Added to MANIFEST: t/pod-coverage.t
Added to MANIFEST: t/pod.t
Added to MANIFEST: xt/boilerplate.t
t/test-dist.t ....... 1/5 Added to MANIFEST: Build.PL
Added to MANIFEST: Changes
Added to MANIFEST: ignore.txt
Added to MANIFEST: lib/Book/Park/Mansfield.pm
Added to MANIFEST: lib/Book/Park/Mansfield/Base.pm
Added to MANIFEST: lib/Book/Park/Mansfield/FannyPrice.pm
Added to MANIFEST: lib/JAUSTEN/Utils.pm
Added to MANIFEST: MANIFEST
Added to MANIFEST: README
Added to MANIFEST: t/00-load.t
Added to MANIFEST: t/manifest.t
Added to MANIFEST: t/pod-coverage.t
Added to MANIFEST: t/pod.t
Added to MANIFEST: xt/boilerplate.t
Argument "^E\n\0" isn't numeric in numeric gt (>) at t/test-dist.t line 1392.
Argument "v5.10.1" isn't numeric in numeric gt (>) at t/test-dist.t line 1392.
Use of uninitialized value in split at /home/litt/.cpan/build/Module-Starter-1.71-dl41ct/blib/lib/Module/Starter/Simple.pm line 1386.
Use of uninitialized value in string eq at t/test-dist.t line 1251.
Use of uninitialized value in hash element at t/test-dist.t line 546.
Use of uninitialized value in hash element at t/test-dist.t line 547.
Use of uninitialized value in string eq at t/test-dist.t line 552.
Use of uninitialized value in string eq at t/test-dist.t line 561.
Use of uninitialized value in concatenation (.) or string at t/test-dist.t line 561.
Use of uninitialized value in hash element at t/test-dist.t line 546.
Use of uninitialized value in hash element at t/test-dist.t line 547.
Use of uninitialized value in string eq at t/test-dist.t line 552.
Use of uninitialized value in string eq at t/test-dist.t line 561.
Use of uninitialized value in concatenation (.) or string at t/test-dist.t line 561.
Use of uninitialized value in hash element at t/test-dist.t line 546.
Use of uninitialized value in hash element at t/test-dist.t line 547.
Use of uninitialized value in string eq at t/test-dist.t line 552.
Use of uninitialized value in string eq at t/test-dist.t line 561.
Use of uninitialized value in concatenation (.) or string at t/test-dist.t line 561.
Use of uninitialized value in hash element at t/test-dist.t line 546.
Use of uninitialized value in hash element at t/test-dist.t line 547.
Use of uninitialized value in string eq at t/test-dist.t line 552.
Use of uninitialized value in string eq at t/test-dist.t line 561.
Use of uninitialized value in concatenation (.) or string at t/test-dist.t line 561.
Use of uninitialized value in hash element at t/test-dist.t line 546.
Use of uninitialized value in hash element at t/test-dist.t line 547.
Use of uninitialized value in string eq at t/test-dist.t line 552.
Use of uninitialized value in string eq at t/test-dist.t line 561.
Use of uninitialized value in concatenation (.) or string at t/test-dist.t line 561.
Use of uninitialized value in string eq at t/test-dist.t line 564.
Use of uninitialized value in string eq at t/test-dist.t line 564.
                            # {
                            #   '__srand' => 3215746354,
                            #   'author' => 'n9X1YtoRbqeqE Z7ajhPM',
                            #   'builder' => undef,
                            #   'dir' => 't/data/loop/XteKfuuh',
                            #   'distro' => 'XteKfuuh',
                            #   'email' => 'V.Z8ezpU.H4Zt.pn_doc2BzHwoPRkXZlg@OWGFeEARVHYRx.KP7rfD9r7.tld',
                            #   'fn' => 't/data/loop/XteKfuuh/Makefile.PL',
                            #   'force' => 0,
                            #   'ignores_type' => [
                            #     'generic',
                            #     'cvs',
                            #     'git',
                            #     'manifest'
                            #   ],
                            #   'license' => undef,
                            #   'minperl' => '5.008001',
                            #   'modules' => [
                            #     'HzChEhFm8bV1UbOHeA6n',
                            #     'lJEeM321bl74',
                            #     'WSx::x',
                            #     'ymWxkK27t1yVRa::cs6qJvknhkNRRyMSuJSP'
                            #   ],
                            #   'verbose' => 0
                            # }
                            # Child (Makefile.PL) exited without calling finalize()

                        #   Failed test 'Makefile.PL'
                        #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
                        # Child (Set ==> HzChEhFm8bV1UbOHeA6n) exited without calling finalize()

                    #   Failed test 'Set ==> HzChEhFm8bV1UbOHeA6n'
                    #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
                    # Child (force = 0) exited without calling finalize()

                #   Failed test 'force = 0'
                #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
                # Child (ignores_type = 0111) exited without calling finalize()

            #   Failed test 'ignores_type = 0111'
            #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
            # Child (minperl = 5.008001) exited without calling finalize()

        #   Failed test 'minperl = 5.008001'
        #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
        # Child (license = mozilla) exited without calling finalize()

    #   Failed test 'license = mozilla'
    #   at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266.
No testing template for Makefile.PL at t/test-dist.t line 1033.
        TestParseFile::parse_file_start(TestParseFile=HASH(0x9743790)) called at t/test-dist.t line 1292
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "Makefile.PL", CODE(0x9743754)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("Makefile.PL", CODE(0x9743754)) called at t/test-dist.t line 1295
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "Set ==> HzChEhFm8bV1UbOHeA6n", CODE(0x974c3b8)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("Set ==> HzChEhFm8bV1UbOHeA6n", CODE(0x974c3b8)) called at t/test-dist.t line 1310
        main::run_settest(ARRAY(0x974c52c), HASH(0x974c454)) called at t/test-dist.t line 1428
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "force = 0", CODE(0x974c91c)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("force = 0", CODE(0x974c91c)) called at t/test-dist.t line 1446
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "ignores_type = 0111", CODE(0x9751910)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("ignores_type = 0111", CODE(0x9751910)) called at t/test-dist.t line 1447
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "minperl = 5.008001", CODE(0x955fbb4)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("minperl = 5.008001", CODE(0x955fbb4)) called at t/test-dist.t line 1448
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "license = mozilla", CODE(0x935c620)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("license = mozilla", CODE(0x935c620)) called at t/test-dist.t line 1449
        main::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 261
        Test::Builder::__ANON__() called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        eval {...} called at /usr/lib/perl5/5.8.8/Test/Builder.pm line 266
        Test::Builder::subtest(Test::Builder=HASH(0x91803d4), "builder = ExtUtils::MakeMaker", CODE(0x93638c4)) called at /usr/lib/perl5/5.8.8/Test/More.pm line 771
        Test::More::subtest("builder = ExtUtils::MakeMaker", CODE(0x93638c4)) called at t/test-dist.t line 1450
    # Child (builder = ExtUtils::MakeMaker) exited without calling finalize()

#   Failed test 'builder = ExtUtils::MakeMaker'
#   at t/test-dist.t line 0.
# Looks like you planned 5 tests but ran 3.
# Looks like you failed 1 test of 3 run.
# Looks like your test exited with 9 just after 3.
t/test-dist.t ....... Dubious, test returned 9 (wstat 2304, 0x900)
Failed 3/5 subtests

Test Summary Report
-------------------
t/test-dist.t     (Wstat: 2304 Tests: 3 Failed: 1)
  Failed test:  3
  Non-zero exit status: 9
  Parse errors: Bad plan.  You planned 5 tests but ran 3.
Files=6, Tests=24,  1 wallclock secs ( 0.05 usr  0.33 sys +  0.28 cusr  0.86 csys =  1.52 CPU)
Result: FAIL
Failed 1/6 test programs. 1/24 subtests failed.
make: *** [test_dynamic] Error 9
  XSAWYERX/Module-Starter-1.71.tar.gz
  /usr/bin/make test -- NOT OK

```

Version info:

```
This is perl, v5.8.8 built for i686-linux-thread-multi
Linux * 2.6.17-1.2142_FC4.stk16 #1 Tue Jul 18 10:19:55 EDT 2006 i686 GNU/Linux
```

--- a/t/test-dist.t
+++ b/t/test-dist.t
@@ -1371,7 +1371,7 @@ my @licenses = keys %$Module::Starter::Simple::LICENSES;
 
 foreach my $builder (qw(ExtUtils::MakeMaker Module::Build Module::Install)) {
 subtest "builder = $builder" => sub {
-    $builder = $builder;
+    $builder = $builder; # 5.8 scoping issue (GH#57)
     undef $@;
     eval "require $builder";  # require hates string class names; must use eval string instead of block
     plan ($@ ? 
@@ -1381,7 +1381,7 @@ subtest "builder = $builder" => sub {
 
     foreach my $license (@licenses) {
     subtest "license = $license" => sub {
-        $license = $license;
+        $license = $license; # 5.8 scoping issue (GH#57)
         plan tests => 5;
 
         foreach my $minperl (5.006, 5.008001, v5.10.0, 'v5.10.1', $^V) {

--- a/t/test-dist.t
+++ b/t/test-dist.t
@@ -1371,6 +1371,7 @@ my @licenses = keys %$Module::Starter::Simple::LICENSES;
 
 foreach my $builder (qw(ExtUtils::MakeMaker Module::Build Module::Install)) {
 subtest "builder = $builder" => sub {
+    $builder = $builder;
     undef $@;
     eval "require $builder";  # require hates string class names; must use eval string instead of block
     plan ($@ ? 
@@ -1380,6 +1381,7 @@ subtest "builder = $builder" => sub {
 
     foreach my $license (@licenses) {
     subtest "license = $license" => sub {
+        $license = $license;
         plan tests => 5;
 
         foreach my $minperl (5.006, 5.008001, v5.10.0, 'v5.10.1', $^V) {

